### PR TITLE
[WFLY-17648] Add new ee-core-profile-server layer and example standal…

### PR DESCRIPTION
**Pull request by jamezp**
…one-core.xml configuration file.

https://issues.redhat.com/browse/WFLY-17648

### CODE REVIEW

File: ee-feature-pack/galleon-common/src/main/resources/layers/standalone/ee-core-profile-server/layer-spec.xml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<layer-spec xmlns="urn:jboss:galleon:layer-spec:1.0" name="ee-core-profile-server">
+    <dependencies>
+        <layer name="core-server"/>
+        <layer name="cdi"/>
+        <layer name="jaxrs"/>
```
Comment (jamezp) [inline ee-feature-pack/galleon-common/src/main/resources/layers/standalone/ee-core-profile-server/layer-spec.xml:25]: Should we be explicit and add the jsonb and jsonp layers even though they're already copied in?

File: ee-feature-pack/galleon-content/src/main/resources/feature_groups/standalone-core.xml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<feature-group-spec name="standalone-core" xmlns="urn:jboss:galleon:feature-group:1.0">
+
+    <feature spec="server-root">
```
Comment (jfdenise) [inline ee-feature-pack/galleon-content/src/main/resources/feature_groups/standalone-core.xml:4]: This feature-group is not included in the ee-core-profile-server layer. Is it really needed?

Comment (jamezp) [reply ee-feature-pack/galleon-content/src/main/resources/feature_groups/standalone-core.xml:4]: TBH I don't know. I just copied what `standalone-full.xml` does and tried to modify what I thought would not be needed.

Comment (jamezp) [reply ee-feature-pack/galleon-content/src/main/resources/feature_groups/standalone-core.xml:4]: Where is `server-root` defined? I can't find anything in core or full where it's actually defined.

Comment (jfdenise) [reply ee-feature-pack/galleon-content/src/main/resources/feature_groups/standalone-core.xml:4]: https://github.com/wildfly/wildfly-core/blob/main/core-feature-pack/galleon-common/src/main/resources/layers/standalone/base-server/layer-spec.xml
that contains standalone-base that contains it. All layers depend on base-server layer, so you will have it provisioned.

File: ee-feature-pack/galleon-common/src/main/resources/layers/standalone/ee-core-profile-server/layer-spec.xml (hunk 1 above)
Comment (jfdenise) [reply ee-feature-pack/galleon-common/src/main/resources/layers/standalone/ee-core-profile-server/layer-spec.xml:25]: I would say that having them explicit would help.

Comment (jamezp) [reply ee-feature-pack/galleon-common/src/main/resources/layers/standalone/ee-core-profile-server/layer-spec.xml:25]: Okay, that was my thinking as well. I'll update that.

File: ee-feature-pack/galleon-content/src/main/resources/feature_groups/standalone-core.xml (hunk 2 above)
Comment (jamezp) [reply ee-feature-pack/galleon-content/src/main/resources/feature_groups/standalone-core.xml]: Perfect, thanks!

**Comment by bstansberry** [comment]
Thanks @jamezp and @jfdenise

**Pull request merged**